<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head>
<title>
DECIPHER - Ancestral States
</title>
<link rel="stylesheet" type="text/css" href="data/theme.css">
<script type="text/javascript" src="data/script.js"></script>
</head>

<body id="Body" onload="numberLines(); toggle();">
<div id="Header"><a href="/"><img class="header" alt="DECIPHER logo" src="images/DECIPHER.png"></a><br clear="all"><br>
<div id="Main">
<ul id="NavBar">

<li><a href="Alignment.html" class="l1">Alignment<span class="right">&#9656;</span></a></li>
<li><a href="Classification.html" class="l1">Classification<span class="right">&#9656;</span></a></li>
<li><a href="Homology.html" class="l1">Homology<span class="right">&#9656;</span></a></li>
<li><a href="OligoDesign.html" class="l1">Oligo Design<span class="right">&#9656;</span></a></li>
<li><a href="Phylogenetics.html" class="l1sub">Phylogenetics<span class="right">&#9662;</span></a></li>
<li><a href="MinimumEvolutionTrees.html" class="l2">Minimum Evolution</a></li>
<li><a href="MaximumLikelihoodTrees.html" class="l2">Maximum Likelihood</a></li>
<li><a href="MaximumParsimonyTrees.html" class="l2">Maximum Parsimony</a></li>
<li><a href="AncestralStates.html" class="l2sel">Ancestral States</a></li>
<li><a href="Tutorials.html" class="l1">Tutorials<span class="right">&#9656;</span></a></li>

</ul>
</div>

<div id="ContentFrame">

<ul id="TopBar">
<li><a href="index.html" class="l1">Home</a></li>
<li><a href="News.html" class="l1">News</a></li>
<li><a href="Downloads.html" class="l1">Downloads</a></li>
<li><a href="Contact.html" class="l1">Contact</a></li>
<li><a href="Citations.html" class="l1">Citations</a></li>
</ul>

<div id="Content">

<p class="c0">Ancestral States</p>

This short example describes how to use <i>Treeline</i> to perform ancestral state reconstruction.  This process results in a tree with estimated states at each internal node.  Maximum likelihood trees infer ancestral states based on likelihood, while the other tree building methods use parsimony.

<br><br>

For an in-depth tutorial on phylogenetics, see the &quot;Growing Phylogenetic Trees with Treeline&quot; vignette, available from the <a href="Documentation.html">Documentation</a> page.

<p class="c1">How do I perform ancestral state reconstruction?</p>

First it is necessary to <a href="Downloads.html">install</a> DECIPHER and load the library in R. After optimizing a tree with <i>reconstruct</i> set to <i>TRUE</i>, it is possible to see the estimated ancestral states or use <i>MapCharacters</i> to tabulate state changes.

<br><br>
<a id="displayText" href="javascript:toggle();">Hide output</a><br>

<pre><span class="codebox"><span class="line"><span class='comments'># load the DECIPHER library in R</span></span>
&gt; <span class='functs'>library</span>&#40;DECIPHER&#41;
&gt; 
&gt; <span class='comments'># load the target sequences from a file</span>
&gt; fas &lt;- <span class='chars'>&quot;&lt;&lt;REPLACE WITH PATH TO FASTA FILE&gt;&gt;&quot;</span>
&gt; seqs &lt;- <span class='functs'>readDNAStringSet</span>&#40;fas&#41; <span class='comments'># use AA, DNA, or RNA</span>
&gt; seqs
DNAStringSet object of length 317:
      width seq               names               
  [1]   819 ATGGCTT...AAGAAAA Rickettsia prowaz...
  [2]   822 ATGGGAA...GAAAAAG Porphyromonas gin...
  [3]   822 ATGGGAA...GAAAAAG Porphyromonas gin...
  [4]   822 ATGGGAA...GAAAAAG Porphyromonas gin...
  [5]   819 ATGGCTA...TGGTAAA Pasteurella multo...
  ...   ... ...
[313]   819 ATGGCAA...TACTAAA Pectobacterium at...
[314]   822 ATGCCTA...CGTCAAG Acinetobacter sp....
[315]   864 ATGGGCA...TCAGTCT Thermosynechococc...
[316]   831 ATGGCAC...GAAGAAG Bradyrhizobium ja...
[317]   840 ATGGGCA...GCGAGGT Gloeobacter viola...
&gt; 
&gt; <span class='comments'># align coding sequences</span>
&gt; seqs &lt;- <span class='functs'>AlignTranslation</span>&#40;seqs,
+ type=<span class='chars'>&quot;DNAStringSet&quot;</span>&#41; <span class='comments'># choose AA or DNA</span>
Determining distance matrix based on shared 5-mers:
  |========================================| 100%
<br>
Time difference of 0.31 secs
<br>
Clustering into groups by similarity:
  |========================================| 100%
<br>
Time difference of 0.02 secs
<br>
Aligning Sequences:
  |========================================| 100%
<br>
Time difference of 0.46 secs
<br>
Iteration 1 of 2:
<br>
Determining distance matrix based on alignment:
  |========================================| 100%
<br>
Time difference of 0.05 secs
<br>
Reclustering into groups by similarity:
  |========================================| 100%
<br>
Time difference of 0.03 secs
<br>
Realigning Sequences:
  |========================================| 100%
<br>
Time difference of 0.21 secs
<br>
Iteration 2 of 2:
<br>
Determining distance matrix based on alignment:
  |========================================| 100%
<br>
Time difference of 0.05 secs
<br>
Reclustering into groups by similarity:
  |========================================| 100%
<br>
Time difference of 0.02 secs
<br>
Realigning Sequences:
  |========================================| 100%
<br>
Time difference of 0.03 secs
<br>
&gt; 
&gt; <span class='comments'># optimize the tree</span>
&gt; tree &lt;- <span class='functs'>Treeline</span>&#40;seqs,
+ method=<span class='chars'>&quot;ME&quot;</span>,
+ model=<span class='chars'>&quot;TN93+F&quot;</span>, <span class='comments'># choose a model</span>
+ reconstruct=TRUE, <span class='comments'># ancestral state reconstuction</span>
+ processors=NULL&#41; <span class='comments'># use all CPUs</span>
Optimizing up to 400 candidate trees:
Tree #68. length = 19.918 (0.000%), 18 Climbs, 0 Grafts of 3  
<br>
Finalizing the best tree (#8):
length = 19.918 (0.000%), 0 Climbs               
<br>
Time difference of 3.65 secs
<br>
&gt; 
&gt; <span class='comments'># examine one ancestral state</span>
&gt; <span class='functs'>attr</span>&#40;tree, <span class='chars'>&quot;state&quot;</span>&#41; <span class='comments'># estimated root state</span>
[1] "GTGAGCTTTTCCCAGAGAAGCTTAGAACAAGGAGGTAAAAAAATGGCAATTAAAAAATTAAAGCCAACTACAAACCCAGGGAGAAGACACAAGACTATTTCCGATTTTGAAGAAATAGAAAAAATCACAAAAACAGACAAAAAATAGGGGAAAAAAAAAAAAAACAAAGTAACACCAGAAAAGTCCCTGCTAGTGCCGATGGTAGCAAAGAAAACAGGAGGACGCAACAGAAATAACGGTAAAATTACCACTCGTCACAAAGGTGGCGGACACAAAAAAAAATACCGAATAATAGATTTCAATAAGAGATACAACCACAAAGACGAAGTTCCTGCAAAAGTAACGGCAATCGAGTACGACCCGAACAGAACTGCAAGGATTGCTCTGCTTCATTACGCTGAAGATGGAGAAAAGAGTTATATTCTCGCTCCCAAAGGTTTGAAAGTGGGCGACACAGTCATAACAGGTGAAAAAGAAGACGCGGAAGCCGGAAAGCCCAAAGCCGAAATCAAACCAGGAAATGCCCTGCCTCTGGAAAACATACCGGTCGGTACCATTATCCACAACATTGAGTTGAATCCTGGAAAGGGTGGACAGATAGCAAGATCTGCCGGAACATATGCCCAGCTTACGGCTAATGACAAAGAAGGAAAATACGCTATGATCAGAATGCCTTCAGGTGAAGTGAGAAAGATACACAACAAGTGCAAGGCCACCATCGGTGAAGTTGGAAACGCAGATCACGAAAACGTAAATCTAGGTAAGGCTGGACGCTCGCGATGGCTAGGTAGCCGACCGCACATCCGTGGTATGGCAATGAACCCGGTTGATCACCCGCTCGGTGGTGGTGAAGGTAGAACGAAATCTGCTAGAGGTCAAAAGCACCCAAAAACTCCTTGGGGACAGCCGACTAAGGGTTACAAGACTAGAAATAATAAGAAACCTTCCAATAAGTTCATCATCAAGAGAAGAAAAAAAAAGAAAAAAAAACAATTGAAACTCCGAAAGCGCGGAGGACGTGAGTCT"
&gt; 
&gt; <span class='comments'># tabulate state changes at each site</span>
&gt; <span class='functs'>head</span>&#40;<span class='functs'>MapCharacters</span>&#40;tree, type=<span class='chars'>&quot;table&quot;</span>&#41;, n=<span class='numbers'>100</span>&#41;
<br>
A306G T567C C258T A573G A174G  A48T A612T A639G 
   29    25    24    22    19    19    19    19 
A892T A935G C228T C273T C285T T420C T792C A616G 
   19    19    19    19    19    19    19    18 
A798G C591T A292G A313G C357T C366T C515T T735C 
   18    18    17    17    17    17    17    17 
A276G A513G  A57G A582G A637C A684G A694G A858T 
   16    16    16    16    16    16    16    16 
A961G C267T C697T C840T T327C A183G A243C A270T 
   16    16    16    16    16    15    15    15 
A326G A370T A399T A456T A471T A603T A738T A974G 
   15    15    15    15    15    15    15    15 
C297T C327T C351T C396T C405T C609T C657T C708T 
   15    15    15    15    15    15    15    15 
C720T C723T C960T T522C T768A A174T A186T A250G 
   15    15    15    15    15    14    14    14 
A289T A321G A442C A454C A457G A462T A470C A501T 
   14    14    14    14    14    14    14    14 
A556G A597G A667C A717T A759T A774T A780T A942G 
   14    14    14    14    14    14    14    14 
C360T  C47G C567T C735T C795T C815T C877A G397A 
   14    14    14    14    14    14    14    14 
T264C T357C A110C A180G A189G A192T A213G A443C 
   14    14    13    13    13    13    13    13 
A454G A499G A553T A568G 
   13    13    13    13</span></pre>

</div>

<div class="fadedScroller_fade"></div>

</div>
</div>
<br />
</body></html>
