<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head>
<title>
DECIPHER - Citations
</title>
<link rel="stylesheet" type="text/css" href="data/theme.css">
<script type="text/javascript" src="data/script.js"></script>
</head>

<body id="Body" onload="numberLines(); toggle();">
<div id="Header"><a href="/"><img class="header" alt="DECIPHER logo" src="images/DECIPHER.png"></a><br clear="all"><br>
<div id="Main">
<ul id="NavBar">

<li><a href="Alignment.html" class="l1">Alignment<span class="right">&#9656;</span></a></li>
<li><a href="Classification.html" class="l1">Classification<span class="right">&#9656;</span></a></li>
<li><a href="Homology.html" class="l1">Homology<span class="right">&#9656;</span></a></li>
<li><a href="OligoDesign.html" class="l1">Oligo Design<span class="right">&#9656;</span></a></li>
<li><a href="Phylogenetics.html" class="l1">Phylogenetics<span class="right">&#9656;</span></a></li>
<li><a href="Tutorials.html" class="l1">Tutorials<span class="right">&#9656;</span></a></li>

</ul>
</div>

<div id="ContentFrame">

<ul id="TopBar">
<li><a href="index.html" class="l1">Home</a></li>
<li><a href="News.html" class="l1">News</a></li>
<li><a href="Downloads.html" class="l1">Downloads</a></li>
<li><a href="Contact.html" class="l1">Contact</a></li>
<li><a href="Citations.html" class="l1sel">Citations</a></li>
</ul>

<div id="Content">

<p class="c0">Citations</p>

<ul>

<li><u>DECIPHER software</u>
<ul>
<li>Versions 3.x:<p class="hanging">ES Wright (2024) "Fast and Flexible Search for Homologous Biological Sequences with DECIPHER v3." <i>The R Journal</i>, <a href="https://journal.r-project.org/articles/RJ-2024-022/RJ-2024-022.pdf">16(2), 191-200</a>.</p></li>
<li>Versions 2.x:<p class="hanging">ES Wright (2016) "Using DECIPHER v2.0 to Analyze Big Biological Sequence Data in R." <i>The R Journal</i>, <a href="https://journal.r-project.org/archive/2016-1/wright.pdf">8(1), 352-359</a>.</p></li>
</ul>
</li>

<li><u>Specific functions</u>
<ul>
<li>DetectRepeats:<p class="hanging">Cho, S.-T. & Wright, E. S. (2025) "Accurate detection of tandem repeats exposes ubiquitous reuse of biological sequences." <i>Nucleic Acids Research</i>, <a href="http://doi.org/10.1093/nar/gkaf866">doi:10.1093/nar/gkaf866</a>.</p></li>
<li>Clusterize:<p class="hanging">ES Wright (2024) "Accurately clustering biological sequences in linear time by relatedness sorting." <i>Nature Communications</i>, <a href="https://www.nature.com/articles/s41467-024-47371-9">doi:10.1038/s41467-024-47371-9</a>.</p></li>
<li>Find Chimeras:<p class="hanging">ES Wright <i>et al.</i> (2012) "DECIPHER, A Search-Based Approach to Chimera Identification for 16S rRNA Sequences." <i>Applied and Environmental Microbiology</i>, <a href="http://aem.asm.org/cgi/pmidlookup?view=long&pmid=22101057">doi:10.1128/AEM.06516-11</a>.</p></li>
</ul>
</li>

<li><u>Sequence alignment</u>
<ul>
<li>Amino acid sequence alignment:<p class="hanging">ES Wright (2015) "DECIPHER: harnessing local sequence context to improve protein multiple sequence alignment." <i>BMC Bioinformatics</i>, <a href="http://www.biomedcentral.com/1471-2105/16/322">doi:10.1186/s12859-015-0749-z</a>.</p></li>
<li>DNA or RNA sequence alignment:<p class="hanging">ES Wright (2020) "RNAconTest: Comparing Tools for Noncoding RNA Multiple Sequence Alignment Based on Structural Consistency." <i>RNA</i>, <a href="https://rnajournal.cshlp.org/content/26/5/531.long">doi:10.1261/rna.073015.119</a>.</p></li>
</ul>
</li>

<li><u>Classification</u>
<ul>
<li>IDTAXA protein (functional) classification:<p class="hanging">NP Cooley &amp; ES Wright (2021) "Accurate annotation of protein coding sequences with IDTAXA." <i>NAR Genomics and Bioinformatics</i>, <a href="https://doi.org/10.1093/nargab/lqab080">doi:10.1093/nargab/lqab080</a>.</p></li>
<li>IDTAXA nucleotide (organismal) classification:<p class="hanging">A Murali <i>et al.</i> (2018) "IDTAXA: a novel approach for accurate taxonomic classification of microbiome sequences." <i>Microbiome</i>, <a href="https://doi.org/10.1186/s40168-018-0521-5">doi:10.1186/s40168-018-0521-5</a>.</p></li>
</ul>
</li>

<li><u>Oligo design</u>
<ul>
<li>Design Signatures:<p class="hanging">ES Wright &amp; KH Vetsigian (2016) "DesignSignatures: a tool for designing primers that yields amplicons with distinct signatures." <i>Bioinformatics</i>, <a href="http://doi.org/10.1093/bioinformatics/btw047">doi:10.1093/bioinformatics/btw047</a>.</p></li>
<li>Design Primers:<p class="hanging">ES Wright <i>et al.</i> (2014) "Exploiting Extension Bias in PCR to Improve Primer Specificity in Ensembles of Nearly Identical DNA Templates." <i>Environmental Microbiology</i>, <a href="http://onlinelibrary.wiley.com/doi/10.1111/1462-2920.12259/abstract">doi:10.1111/1462-2920.12259</a>.</p></li>
<li>Design Probes and ProbeMelt (FISH):<p class="hanging">ES Wright <i>et al.</i> (2014) "Automated Design of Probes for rRNA-Targeted Fluorescence In Situ Hybridization Reveals the Advantages of Using Dual Probes for Accurate Identification." <i>Applied and Environmental Microbiology</i>, <a href="http://aem.asm.org/content/80/16/5124.full">doi:10.1128/AEM.01685-14</a>.</p></li>
</ul>
</li>

</ul>

</div>

<div class="fadedScroller_fade"></div>

</div>
</div>
<br />
</body></html>
