<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head>
<title>
DECIPHER - Examples Gallery
</title>
<link rel="stylesheet" type="text/css" href="data/theme.css">
<script type="text/javascript" src="data/script.js"></script>
</head>

<body id="Body" onload="numberLines(); toggle();">
<div id="Header"><a href="/"><img class="header" alt="DECIPHER logo" src="images/DECIPHER.png"></a><br clear="all"><br>
<div id="Main">
<ul id="NavBar">

<li><a href="Alignment.html" class="l1">Alignment<span class="right">&#9656;</span></a></li>
<li><a href="Classification.html" class="l1">Classification<span class="right">&#9656;</span></a></li>
<li><a href="Homology.html" class="l1">Homology<span class="right">&#9656;</span></a></li>
<li><a href="OligoDesign.html" class="l1">Oligo Design<span class="right">&#9656;</span></a></li>
<li><a href="Phylogenetics.html" class="l1">Phylogenetics<span class="right">&#9656;</span></a></li>
<li><a href="Tutorials.html" class="l1sub">Tutorials<span class="right">&#9662;</span></a></li>
<li><a href="Gallery.html" class="l2sel">Examples Gallery</a></li>
<li><a href="Documentation.html" class="l2">Documentation</a></li>
<li><a href="RLessons.html" class="l2">R Lessons</a></li>
<li><a href="Bioinformatics.html" class="l2">Bioinformatics</a></li>

</ul>
</div>

<div id="ContentFrame">

<ul id="TopBar">
<li><a href="index.html" class="l1">Home</a></li>
<li><a href="News.html" class="l1">News</a></li>
<li><a href="Downloads.html" class="l1">Downloads</a></li>
<li><a href="Contact.html" class="l1">Contact</a></li>
<li><a href="Citations.html" class="l1">Citations</a></li>
</ul>

<div id="Content">

<p class="c0">Examples Gallery</p>

This gallery showcases the <b>DECIPHER package for R</b> in action.<img src="images/Screenshots/Example_Code.png" style="overflow: hidden; margin-left:-10px; margin-top:+10px; margin-right:-15px;" alt="" width="70%" align="right"><br><br>To install and start using the DECIPHER software package for <a href="http://www.r-project.org/">R</a>, visit the <a href="/Downloads.html">Downloads</a> page. There you will find several vignettes that describe how to use DECIPHER within the R environment to accomplish various tasks.  The R programming language provides access to many powerful tools, making it well suited for a wide variety of analyses.

<br><br><img src="images/Screenshots/Example1.png" alt="" width="590px" align="center"><br>Using the <i>BrowseSeqs</i> function to view sequences in a browser (<a href="images/Screenshots/myXStringSet.html" target="_blank">full page</a>, <a href="Example1.html">example code</a>).

<br><br><hr>

<br><img src="images/Screenshots/Example2.png" alt="" width="590px" align="center"><br><br>Plotting the output of the <i>MeltDNA</i> function for two alleles (<a href="Example2.html">example code</a>). The lines represent the predicted melt curves, which could be experimentally verified with Quantitative PCR.

<br><br><hr>

<br><img src="images/Screenshots/Example3.png" alt="" width="590px" align="center"><br><br>Creating a maximum likelihood tree of different <i>Streptomyces</i> species with the <i>Treeline</i> function (<a href="Example3.html">example code</a>).

<br><br><img src="images/Screenshots/Treeline_output.png" alt="" width="590px" align="center"><br>DECIPHER's outputs when generating the phylogenetic tree above.

<br><br><hr>

<br><img src="images/Screenshots/Example4.png" alt="" width="590px" align="center"><br>Maintaining and adding to a database of sequences (<a href="images/Screenshots/db.html" target="_blank">full page</a>, <a href="Example4.html">example code</a>). Each row provides information for a unique sequence in the database.

<br><br><hr>

<br><img src="images/Screenshots/Example5.png" alt="" width="590px" align="center"><br><br>Predicting the melt curves for two FISH probes (<a href="Example5.html">example code</a>). Probe 2 melts at a less stringent condition than Probe 1.

<br><br><hr>

<br><img src="images/Screenshots/Example6dots.png" alt="" width="590px" align="center"><br><br>Dot plots showing the syntenic blocks among multiple <i>Yersinia</i> genomes, as identified by the <i>FindSynteny</i> function (<a href="Example6.html">example code</a>). Black diagonal lines show syntenic regions sharing the same orientation, whereas red anti-diagonal lines represent blocks of synteny between opposite strands.

<br><br><img src="images/Screenshots/Example6bars.png" alt="" width="590px" align="center"><br><br>Bar plots providing an alternative representation of the synteny map shown above. Each genome is colored according to the position of the corresponding region in the first genome, or grey if a region is unshared.

<br><br><hr>

<br><img src="images/Screenshots/Example7.png" alt="" width="590px" align="center"><br><br>Estimated dN/dS (&omega;) ratios for every codon in the Influenza A hemagglutinin protein coding sequence (<a href="Example7.html">example code</a>).

<br><br><hr>

<br><img src="images/Screenshots/Example8.png" alt="" width="590px" align="center"><br><br>Starburst diagram showing the predicted taxonomic classification for a set of 16S ribosomal RNA sequences (<a href="Example8.html">example code</a>).

</div>

<div class="fadedScroller_fade"></div>

</div>
</div>
<br />
</body></html>
