<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head>
<title>
DECIPHER - ProbeMelt
</title>
<link rel="stylesheet" type="text/css" href="data/theme.css">
<script type="text/javascript" src="data/script.js"></script>
</head>

<body id="Body" onload="numberLines(); toggle();">
<div id="Header"><a href="/"><img class="header" alt="DECIPHER logo" src="images/DECIPHER.png"></a><br clear="all"><br>
<div id="Main">
<ul id="NavBar">

<li><a href="Alignment.html" class="l1">Alignment<span class="right">&#9656;</span></a></li>
<li><a href="Classification.html" class="l1">Classification<span class="right">&#9656;</span></a></li>
<li><a href="Homology.html" class="l1">Homology<span class="right">&#9656;</span></a></li>
<li><a href="OligoDesign.html" class="l1sub">Oligo Design<span class="right">&#9662;</span></a></li>
<li><a href="DesignSignatures.html" class="l2">Design Signatures</a></li>
<li><a href="DesignPrimers.html" class="l2">Design Primers</a></li>
<li><a href="DesignProbes.html" class="l2">Design Probes</a></li>
<li><a href="ProbeMelt.html" class="l2sel">ProbeMelt</a></li>
<li><a href="Phylogenetics.html" class="l1">Phylogenetics<span class="right">&#9656;</span></a></li>
<li><a href="Tutorials.html" class="l1">Tutorials<span class="right">&#9656;</span></a></li>

</ul>
</div>

<div id="ContentFrame">

<ul id="TopBar">
<li><a href="index.html" class="l1">Home</a></li>
<li><a href="News.html" class="l1">News</a></li>
<li><a href="Downloads.html" class="l1">Downloads</a></li>
<li><a href="Contact.html" class="l1">Contact</a></li>
<li><a href="Citations.html" class="l1">Citations</a></li>
</ul>

<div id="Content">

<p class="c0">ProbeMelt</p>

DECIPHER contains multiple functions for predicting the effectiveness of oligonucleotides, as described in:<br>

<ul>

<li>FISH probes (<i>CalculateEfficiencyFISH</i>):<p class="hanging">ES Wright <i>et al.</i> (2014) "Automated Design of Probes for rRNA-Targeted Fluorescence In Situ Hybridization Reveals the Advantages of Using Dual Probes for Accurate Identification." <i>Applied and Environmental Microbiology</i>, <a href="http://aem.asm.org/content/80/16/5124.full">doi:10.1128/AEM.01685-14</a>.</p></li>

<li>Primers (<i>CalculateEfficiencyPCR</i>):<p class="hanging">ES Wright <i>et al.</i> (2014) "Exploiting Extension Bias in PCR to Improve Primer Specificity in Ensembles of Nearly Identical DNA Templates." <i>Environmental Microbiology</i>, <a href="http://onlinelibrary.wiley.com/doi/10.1111/1462-2920.12259/abstract">doi:10.1111/1462-2920.12259</a>.</p></li>

</ul>

For an in-depth tutorial on oligonucleotide design, see the associated vignettes on the <a href="Documentation.html">Documentation</a> page.

<p class="c1">How do I calculate hybridization efficiency?</p>

First it is necessary to <a href="Downloads.html">install</a> DECIPHER and load the library in R. Next, specify the oligonucleotide probe or primers and their target sequences:<br><br>

<a id="displayText" href="javascript:toggle();">Hide output</a><br>

<pre><span class="codebox"><span class="line"><span class='functs'>library</span>&#40;DECIPHER&#41;</span>
&gt; 
&gt; oligos &lt;- <span class='functs'>c</span>&#40;<span class='chars'>&quot;TCACCTGGCCATGTCGC&quot;</span>, <span class='chars'>&quot;TCACCTGATCATGTCGC&quot;</span>&#41;
&gt; <span class='comments'># reverse complement of first oligo:</span>
&gt; targets &lt;- <span class='functs'>rep</span>&#40;<span class='chars'>&quot;GCGACATGGCCAGGTGA&quot;</span>, <span class='numbers'>2</span>&#41;
&gt; 
&gt; <span class='functs'>CalculateEfficiencyFISH</span>&#40;oligos, targets,
+ temp=<span class='numbers'>46</span>, <span class='comments'># hybridization temperature</span>
+ P=<span class='numbers'>250e-9</span>, <span class='comments'># probe concentration &#40;molar&#41;</span>
+ ions=<span class='numbers'>1</span>, <span class='comments'># molar Na+ concentration</span>
+ FA=<span class='numbers'>35</span>&#41; <span class='comments'># formamide concentration &#40;% v/v&#41;</span>
      HybEff       FAm     ddG1       dG1
1 0.56553679 37.568501 0.000000 -12.08644
2 0.04949342  6.211841 4.078027 -10.04512
&gt; 
&gt; <span class='functs'>CalculateEfficiencyPCR</span>&#40;oligos, targets,
+ temp=<span class='numbers'>64</span>, <span class='comments'># annealing temperature &#40;Celsius&#41;</span>
+ P=<span class='numbers'>4e-7</span>, <span class='comments'># primer concentration &#40;molar&#41;</span>
+ ions=<span class='numbers'>0.225</span>, <span class='comments'># Na+ equivalent concentration &#40;molar&#41;</span>
+ taqEfficiency=FALSE&#41; <span class='comments'># TRUE if using Taq polymerase</span>
[1] 0.8868337419 0.0005365686</span></pre>

</div>

<div class="fadedScroller_fade"></div>

</div>
</div>
<br />
</body></html>
