<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head>
<title>
DECIPHER - Video #4
</title>
<link rel="stylesheet" type="text/css" href="data/theme.css">
<script type="text/javascript" src="data/script.js"></script>
</head>

<body id="Body" onload="numberLines(); toggle();">
<div id="Header"><a href="/"><img class="header" alt="DECIPHER logo" src="images/DECIPHER.png"></a><br clear="all"><br>
<div id="Main">
<ul id="NavBar">

<li><a href="Alignment.html" class="l1">Alignment<span class="right">&#9656;</span></a></li>
<li><a href="Classification.html" class="l1">Classification<span class="right">&#9656;</span></a></li>
<li><a href="Homology.html" class="l1">Homology<span class="right">&#9656;</span></a></li>
<li><a href="OligoDesign.html" class="l1">Oligo Design<span class="right">&#9656;</span></a></li>
<li><a href="Phylogenetics.html" class="l1">Phylogenetics<span class="right">&#9656;</span></a></li>
<li><a href="Tutorials.html" class="l1sub">Tutorials<span class="right">&#9662;</span></a></li>
<li><a href="Gallery.html" class="l2">Examples Gallery</a></li>
<li><a href="Documentation.html" class="l2">Documentation</a></li>
<li><a href="RLessons.html" class="l2">R Lessons</a></li>
<li><a href="Bioinformatics.html" class="l2sel">Bioinformatics</a></li>

</ul>
</div>

<div id="ContentFrame">

<ul id="TopBar">
<li><a href="index.html" class="l1">Home</a></li>
<li><a href="News.html" class="l1">News</a></li>
<li><a href="Downloads.html" class="l1">Downloads</a></li>
<li><a href="Contact.html" class="l1">Contact</a></li>
<li><a href="Citations.html" class="l1">Citations</a></li>
</ul>

<div id="Content">

<p class="c0">Predicting gene functional associations from coevolutionary signals with EvoWeaver</p>

Originally presented at the <i>ISMB 2024</i> conference in Montreal, Canada.<br><br>

<!--<video width="590" controls><source src="data/Videos/Video4.mp4" type="video/mp4">Your browser does not support this video format.</video>-->
<iframe width="590" height="443" src="https://drive.google.com/file/d/1F4LMC9FM9B4WWpiGW1MHwA2tbrpgXYzs/preview"></iframe>

<br><br>The universe of uncharacterized proteins is expanding far faster than our ability to annotate their functions through laboratory study. Computational annotation approaches rely on similarity to previously studied proteins, thereby ignoring unstudied proteins. This phenomenon gives rise to a "rich get richer" scenario: the majority of research focuses on a small subset of proteins. Coevolutionary approaches hold promise for injecting new information into our knowledge of the protein universe by linking proteins through 'guilt-by-association'. However, existing coevolutionary algorithms have insufficient accuracy and scalability to connect the entire universe of proteins. We present EvoWeaver, an algorithm that weaves together 12 distinct signals of coevolution to quantify the degree of shared evolution between genes. EvoWeaver's signals encompass phylogenetic profiling, phylogenetic structure, gene organization, and sequence-level methods that broadly capture coevolution between sequences. EvoWeaver accurately identifies proteins involved in protein complexes or separate steps of a biochemical pathway. We demonstrate the merits of EvoWeaver by partly reconstructing known biochemical pathways without any prior knowledge other than genome sequences. Additionally, we show that EvoWeaver's predictions rival those of the widely used STRING database without reliance on prior biological knowledge. Finally, we leverage EvoWeaver's predictions to uncover experimentally validated functional associations among genes that are absent from existing databases. This work forms one of the largest scale analyses of protein functional relationships to date, encompassing 1,545 gene groups from 8,564 genomes. Given its predictive power and speed, EvoWeaver has the potential to revolutionize protein functional prediction at the scale of the protein universe.<br><br>

</div>

<div class="fadedScroller_fade"></div>

</div>
</div>
<br />
</body></html>
